File: Day4/round1_p2.java
// 2022 IUSB Programming Competition
// Round 1 Problem 2
// Credit Card Number
// Solution by Liguo Yu

import java.util.Scanner;

public class round1_p2
{	
    public static void main(String[] args)
    {
    	Scanner input = new Scanner(System.in);
        
        int total = 0;
        String s = input.nextLine();
    	
        for(int i=0; i<16; i++)
        {
            char c = s.charAt(i);
            total = total + Integer.parseInt(String.valueOf(c)); 
        }
        
        if(total%29 ==5) System.out.println("valid");
        else  System.out.println("not valid");	
    }
}

File: Day4/round1_p2.py
# 2022 IUSB Programming Competition
# Round 1 Problem 2
# Credit Card Number
# Solution by Janet Yu (high school volunteer)
# Uses Python 3.9.5

if __name__ == '__main__':
	number = input()
	sum = 0

	for x in number:
		a = int(x)
		sum += a

	if sum%29 == 5:
		print("valid")
	else:
		print("not valid")